dask_mongo/__init__.py

```diff
@@ -1,2 +1,2 @@
 from ._version import __version__
-from .core import read_mongo, to_mongo
+from .core import read_mongo, to_mongo, _get_num_clients, _get_client, _CACHE_SIZE
```

Comment:
I don't think we want users using these private methods. Given that, I'm not sure why we would add them to the top-level `dask_mongo` namespace (though I certainly could be missing something here) 

```suggestion
from .core import read_mongo, to_mongo
```